core/src/main/scala/kanaloa/reactive/dispatcher/queue/QueueProcessor.scala

```diff
@@ -13,158 +10,151 @@ import kanaloa.reactive.dispatcher.{Backend, ResultChecker}
 import kanaloa.util.FiniteCollection._
 import kanaloa.util.MessageScheduler
 
+import util.{Failure, Success}
 import scala.concurrent.duration._
 
 trait QueueProcessor extends Actor with ActorLogging with MessageScheduler {
   import QueueProcessor.WorkerPool
+
   val queue: QueueRef
   def backend: Backend
   def settings: ProcessingWorkerPoolSettings
   def resultChecker: ResultChecker
   val metricsCollector: MetricsCollector
+  def workerFactory: WorkerFactory
   type ResultHistory = Vector[Boolean]
   val resultHistoryLength: Int
   protected def onWorkError(resultHistory: ResultHistory, pool: WorkerPool)
+  var resultHistory = Vector[Boolean]()
+  var workerPool = Set[ActorRef]()
 
   metricsCollector.send(Metric.PoolSize(settings.startingPoolSize))
 
-  override val supervisorStrategy =
-    OneForOneStrategy(maxNrOfRetries = 10, withinTimeRange = 1.minute) {
-      case _: Exception ⇒ Restart
-    }
-
-  protected def workerProp(queueRef: QueueRef): Props = Worker.default(queue, backend)(resultChecker)
+  //stop any children which failed.  Let the DeathWatch handle it
+  override val supervisorStrategy = SupervisorStrategy.stoppingStrategy
 
-  def receive: Receive = {
-    val workers = (1 to settings.startingPoolSize).map(createWorker).toSet
+  override def preStart(): Unit = {
+    super.preStart()
+    (1 to settings.startingPoolSize).foreach(x ⇒ createRoutee())
     settings.maxProcessingTime.foreach(delayedMsg(_, QueueMaxProcessTimeReached(queue)))
     context watch queue
-
-    monitoring()(workers)
   }
 
-  def monitoring(resultHistory: ResultHistory = Vector.empty)(pool: WorkerPool): Receive = {
-    def workError(): Unit = {
-      val newHistory = resultHistory.enqueueFinite(false, resultHistoryLength)
-      context become monitoring(newHistory)(pool)
-      onWorkError(newHistory, pool)
-    }
+  def workError(): Unit = {
```

Comment:
now that it's a class scope method, it might be better to rename it to recordWorkError() to be more clear?